tests/cpptests/test_cpp_query.cpp

```diff
@@ -181,6 +184,8 @@ TEST_F(QueryTest, testParser) {
   assertValidQuery("@tags:{bar* | foo}", ctx);
   assertValidQuery("@tags:{bar* | foo*}", ctx);
 
+  assertInvalidQuery("@title:{{foo}}}}}", ctx);
```

Comment:
Just confirming: `@title:{{foo}}}}}` was legal before this PR and now it will throw an error,
So it's a breaking change but it seems OK.

Reply:
actually my mistake, `@title:{{foo}}}}}` was not valid but `@title:{foo}}}}}` was